BootloaderCorePkg/Tools/GenContainer.py

```diff
@@ -592,12 +601,13 @@ def gen_container_bin (container_list, out_dir, inp_dir, key_dir = '.', tool_dir
         out_file = container.create (each)
         print ("Container '%s' was created successfully at:  \n  %s" % (container.header.signature.decode(), out_file))
 
-def gen_layout (comp_list, img_type, out_file, key_file):
+def gen_layout (comp_list, img_type, sign_hash_alg, out_file, key_file):
+    sign_hash_alg = get_key_type(key_file)
     # prepare the layout from individual components from '-cl'
     if img_type not in CONTAINER_HDR._image_type.keys():
         raise Exception ("Invalid Container Type '%s' !" % img_type)
-    layout = "('BOOT', '%s', '%s', 'RSA2048' , '%s', 0x10, 0),\n" % (out_file, img_type, key_file)
-    end_layout = "('_SG_', '', 'Dummy', 'SHA2_256', '', 0, 0),"
+    layout = "('BOOT', '%s', '%s', '%s' , '%s', 0x10, 0),\n" % (out_file, img_type, sign_hash_alg, key_file)
+    end_layout = "('_SG_', '', 'Dummy', '%s', '', 0, 0)," %(sign_hash_alg)

```

Comment:
sign_hash_alg comes from default parameter SHA2_256, should be changed to to RSA2048 to align this change.

Reply:
Thanks Guo. I am using same variable for key and hash. I would need to discuss further to fix.